src/pokemon_summary_screen.c

```diff
@@ -0,0 +1,3679 @@
+	mov r0, r9\n\
+	lsls r2, r0\n\
+	mov r3, r10\n\
+	lsls r1, r3\n\
+	adds r2, r1\n\
+	ldrb r0, [r5]\n\
+	orrs r0, r2\n\
+	strb r0, [r5]\n\
+	adds r1, r4, 0\n\
+	adds r1, 0xD\n\
+	adds r0, r7, 0\n\
+	mov r2, sp\n\
+	bl SetBoxMonData\n\
+	adds r1, r6, 0\n\
+	adds r1, 0xD\n\
+	adds r0, r7, 0\n\
+	ldr r2, [sp, 0x1C]\n\
+	bl SetBoxMonData\n\
+	adds r4, 0x11\n\
+	adds r0, r7, 0\n\
+	adds r1, r4, 0\n\
+	ldr r2, [sp, 0x20]\n\
+	bl SetBoxMonData\n\
+	adds r6, 0x11\n\
+	adds r0, r7, 0\n\
+	adds r1, r6, 0\n\
+	ldr r2, [sp, 0x24]\n\
+	bl SetBoxMonData\n\
+	adds r0, r7, 0\n\
+	movs r1, 0x15\n\
+	adds r2, r5, 0\n\
+	bl SetBoxMonData\n\
+	mov r0, sp\n\
+	ldrh r0, [r0]\n\
+	ldr r1, [sp, 0x8]\n\
+	strh r0, [r1]\n\
+	ldr r3, [sp, 0x1C]\n\
+	ldrh r0, [r3]\n\
+	ldr r1, [sp, 0xC]\n\
+	strh r0, [r1]\n\
+	ldr r3, [sp, 0x20]\n\
+	ldrb r0, [r3]\n\
+	ldr r1, [sp, 0x10]\n\
+	strb r0, [r1]\n\
+	ldr r3, [sp, 0x24]\n\
+	ldrb r0, [r3]\n\
+	ldr r1, [sp, 0x14]\n\
+	strb r0, [r1]\n\
+	ldrb r0, [r5]\n\
+	ldr r3, [sp, 0x18]\n\
+	strb r0, [r3]\n\
+	add sp, 0x28\n\
+	pop {r3-r5}\n\
+	mov r8, r3\n\
+	mov r9, r4\n\
+	mov r10, r5\n\
+	pop {r4-r7}\n\
+	pop {r0}\n\
+	bx r0\n\
+	.pool\n\
+    .syntax divided\n");
+}
+#endif
+
+void sub_81C171C(u8 taskId)
+{
+    sub_81C44F0();
+    sub_81C4AF8(8);
+    gTasks[taskId].func = sub_81C174C;
+}
+
+void sub_81C174C(u8 taskId)
+{
+    s16* data = gTasks[taskId].data;
+    
+    if (sub_81221EC() != 1)
+    {
+        if (gPaletteFade.active != 1)
+        {
+            if (gMain.newKeys & DPAD_UP)
+            {
+                data[0] = 4;
+                sub_81C1070(data, -1, &gUnknown_0203CF1C->unk40C6);
+            }
+            else if (gMain.newKeys & DPAD_DOWN)
+            {
+                data[0] = 4;
+                sub_81C1070(data, 1, &gUnknown_0203CF1C->unk40C6);
+            }
+            else if (gMain.newKeys & DPAD_LEFT || GetLRKeysState() == 1)
+            {
+                sub_81C0A8C(taskId, -1);
+            }
+            else if (gMain.newKeys & DPAD_RIGHT || GetLRKeysState() == 2)
+            {
+                sub_81C0A8C(taskId, 1);
+            }
+            else if (gMain.newKeys & A_BUTTON)
+            {
+                if (sub_81C18A8() == 1)
+                {
+                    sub_81C48F0();
+                    PlaySE(SE_SELECT);
+                    gUnknown_0203CF21 = gUnknown_0203CF1C->unk40C6;
+                    gSpecialVar_0x8005 = gUnknown_0203CF21;
+                    sub_81C044C(taskId);
+                }
+                else
+                {
+                    PlaySE(0x20);
+                    sub_81C18F4(taskId);
+                }
+            }
+            else if (gMain.newKeys & B_BUTTON)
+            {
+                u32 var1;
+                sub_81C48F0();
+                PlaySE(SE_SELECT);
+                gUnknown_0203CF21 = 4;
+                gSpecialVar_0x8005 = 4;
+                sub_81C044C(taskId);
+            }   
+        }
+    }
+}
+
+u8 sub_81C18A8()
```

Comment:
Seems like this should be a bool8 and the return value should be TRUE and FALSE.